test/test-notification-model-updates.js

```diff
@@ -0,0 +1,382 @@
+        [
+          time, '/xyzzy', 13.092878, 0,
+          {closed: true, mysql: 2.351871, mongodb: 1.623075},
+        ], [
+          time, '/xyzzy', 13.092878, 0,
+          {closed: true, mysql: 2.351871, mongodb: 1.623075},
+        ]
+      ]
+    };
+
+    var notification = {
+      cmd: 'agent:trace',
+      processId: 1235,
+      trace: traceRecord
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  }
+
+  t.test('Notify trace with current data',
+    notifyTraces.bind(this, 0));
+  t.test('Notify trace with stale data',
+    notifyTraces.bind(this, 10 * 60 * 1000));
+
+  t.test('Check worker 1 traces', function(tt) {
+    app.models.AgentTrace.find({}, function(err, traces) {
+      tt.ok(!err, 'traces should be found');
+      tt.equal(traces.length, 2, 'only two traces should be present');
+      var trace = traces[0];
+      tt.equal(trace.processId, 2, 'process id should match');
+      tt.ok(!!trace.trace, 'trace should be populated');
+      tt.end();
+    });
+  });
+
+
+  function notifyExpressRecord(dateBy, tt) {
+    var rec = {
+      timeStamp: Date.now() - dateBy,
+      client: {address: '::1'},
+      request: {method: 'GET', url: '/'},
+      response: {status: 404, duration: 6},
+      process: {pid: 1235},
+      data: {custom: 'value'}
+    };
+
+    var notification = {
+      cmd: 'express:usage-record',
+      record: rec
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  }
+
+  t.test('Notify express usage records with current data',
+    notifyExpressRecord.bind(this, 0));
+  t.test('Notify express usage records with stale data',
+    notifyExpressRecord.bind(this, 10 * 60 * 1000));
+
+  t.test('Check worker 1 express usage records', function(tt) {
+    app.models.ExpressUsageRecord.find({}, function(err, records) {
+      tt.ok(!err, 'express usage records should be found');
+      tt.equal(records.length, 1, 'only one record should be present');
+      var record = records[0];
+      tt.equal(record.processId, 2, 'process id should match');
+      tt.equal(record.workerId, 1);
+      tt.ok(!!record.detail, 'Usage detail is present');
+      tt.end();
+    });
+  });
+
+  t.test('Emit exit', function(tt) {
+    var notification = {
+      cmd: 'exit',
+      pid: 1234,
+      id: 0,
+      reason: 'foo',
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  });
+
+  t.test('Check process exit state', function(tt) {
+    var ServiceProcess = app.models.ServiceProcess;
+    ServiceProcess.find({where: {workerId: 0}}, function(err, procs) {
+      tt.ok(!err, 'process records should be found');
+      tt.equal(procs.length, 1, 'only one process should be present');
+      var proc = procs[0];
+      tt.equal(proc.stopReason, 'foo', 'stop reason should match');
+      tt.ok(!!proc.stopTime, 'stop time should be set');
```

Comment:
I haven't read line-by-line, but would it be possible to test `rpoc.stopTime` against a range or format instead of just asserting that it is truthy?